module/translation-deepl/src/Infrastructure/Cache/DatabaseTranslationCache.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See license.txt for license details.
+ */
+
+declare(strict_types = 1);
+
+namespace Ergonode\TranslationDeepl\Infrastructure\Cache;
+
+use Doctrine\DBAL\Connection;
+use Ergonode\Core\Infrastructure\Cache\CacheInterface;
+use Ergonode\TranslationDeepl\Infrastructure\Provider\TranslationDeeplProvider;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ */
+class DatabaseTranslationCache implements CacheInterface
+{
+    private const TABLE = 'translation_cache';
+    private const FIELDS = [
+        'a.translation',
+    ];
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * TranslationDeeplProviderDecorator constructor.
+     *
+     * @param TranslationDeeplProvider $provider
+     * @param Connection               $connection
+     */
+    public function __construct(TranslationDeeplProvider $provider, Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get(UuidInterface $key)
+    {
+        return $this->connection->createQueryBuilder()
```

Comment:
if there is no cached data this method will return false, should return null or throw exception